libnmstate/schemas/operational-state.yaml

```diff
@@ -408,3 +416,49 @@ definitions:
                     type: boolean
                   state:
                     type: string
+  rib:
+    type: object
+    properties:
+      name:
+        type: string
+      route-table-id:
+        type: integer
+      address-family:
+        type: string
+        enum:
+          - ipv4
+          - ipv6
+      default-rib:
+        type: boolean
+      routes:
+        type: array
+        items:
+          type: object
+          $ref: "#/definitions/route"
+  route:
+    type: object
+    properties:
+      source-protocol:
+        type: string
+        enum:
+          - direct
+          - static
+          - dhcp        # DHCP RFC 3442
+          - ipv6-ra     # IPv6 Router Advertisement
```

Comment:
Don't we need one that contains both the dynamic ones? I remember you mentioned that we can not always differentiate who is the source.

Reply:
The DHCPv6 cannot set route.
For ipv4, it's `dhcp` or `static`.
For ipv6, it's `auto`(ipv6-ra) or `static`.